src/services/WorkPeriodPaymentService.js

```diff
@@ -213,27 +196,29 @@ createWorkPeriodPayment.schema = Joi.object().keys({
  */
 async function updateWorkPeriodPayment (currentUser, id, data) {
   // check permission
-  await _checkUserPermissionForCRUWorkPeriodPayment(currentUser)
+  _checkUserPermissionForCRUWorkPeriodPayment(currentUser)
 
-  if (data.workPeriodId) {
-    // ensure work period exists
-    await helper.ensureWorkPeriodById(data.workPeriodId)
-  }
   const workPeriodPayment = await WorkPeriodPayment.findById(id)
   const oldValue = workPeriodPayment.toJSON()
-
-  data.updatedBy = await helper.getUserId(currentUser.userId)
-  let updated = null
-  try {
-    updated = await workPeriodPayment.update(data)
-  } catch (err) {
-    if (!_.isUndefined(err.original)) {
-      throw new errors.BadRequestError(err.original.detail)
-    } else {
-      throw err
+  if (oldValue.status === data.status) {
+    return oldValue
+  }
+  if (data.status === 'cancelled' && oldValue.status === 'in-progress') {
+    throw new errors.BadRequestError('You cannot cancel a WorkPeriodPayment which is in-progress')
+  }
+  if (data.status === 'scheduled') {
+    if (oldValue.status !== 'failed') {
+      throw new errors.BadRequestError(`You cannot schedule a WorkPeriodPayment which is ${oldValue.status}`)
+    }
```

Comment:
Sorry, I've missed one more condition. We can also schedule the `pending` payments - which is a normal workflow.